lib/sidebar.rb

```diff
@@ -29,178 +30,111 @@ def pre_render(page)
-        subsubgroup_key.include? key and subsubgroup_key != key and \
-          key.split('/').length > level
-      end
-
-      subgroup = ''
-      has_subgroups = !all_subgroups.empty?
-      if value[:headers].any? || !subsubgroup_list.empty?
-        if has_subgroups
-          url = value[:url]
-          active = active?(filename, url, true)
-          # puts "#{url}, #{filename}, #{key}" if active
-          item_class = active || (url.split('/').length > level && filename.start_with?(url)) ? 'nav-subgroup active' : 'nav-subgroup'
-          subgroup << "<li class=\"#{item_class}\">"
-          subgroup << "<div class=\"nav-subgroup-heading\"><i class=\"material-icons\">arrow_right</i><a href=\"#{url}\">#{title}</a></div>"
-          subgroup << '<ul class="nav-ul">'
-
-          if subsubgroup_list.empty?
-            value[:headers].each do |header|
-              subgroup << "<li class=\"nav-leaf\"><a href=\"#{value[:url]}#{header[:hash]}\">#{header[:title]}</a></li>"
-            end
-          else
-            subgroup_leaf_class = active ? 'nav-leaf nav-subgroup-leaf active' : 'nav-leaf nav-subgroup-leaf'
-            subgroup << "<li class=\"#{subgroup_leaf_class}\"><a href=\"#{value[:url]}\">#{title} overview</a></li>"
-
-            subsubgroup_list.each do |subsubgroup_key, subsubgroup_value|
-              subgroup << generateSubgroup(filename, subsubgroup_key, subsubgroup_value, subsubgroup_list, 3)
-            end
-          end
-
-          subgroup << '</ul>'
-          subgroup << '</li>'
-        else
-          value[:headers].each do |header|
-            subgroup << "<li class=\"nav-leaf\"><a href=\"#{value[:url]}#{header[:hash]}\">#{header[:title]}</a></li>"
-          end
-        end
-      else
-        subgroup << if has_subgroups
-                      "<li class=\"nav-leaf nav-subgroup-leaf\"><a href=\"#{value[:url]}\">#{title}</a></li>"
-                    else
-                      "<li class=\"nav-leaf\"><a href=\"#{value[:url]}\">#{title}</a></li>"
-                    end
+        child_markup = header.last_element_child
+        header = {
+          id: header['id'],
+          title: header.content.strip,
+          hash: (child_markup['href']).to_s
+        }
+        headers.push(header)
       end
 
-      subgroup
+      headers
     end
 
-    def render(filename)
-      sidebar = ''
-
-      merged = merge(@hash_pre_render, @filename_with_headers).sort_by { |_key, value| value[:menu_order] }
-      merged.select { |key, _value| key.split('/').length <= 2 }.each do |key, value|
-        next if value[:title].nil?
-        next if value[:hide_from_sidebar]
-
-        subgroups = merged.select { |subgroup_key, _subgroup_value| subgroup_key.include? key and subgroup_key != key and key != '/' }
-
-        active = active?(filename, key)
-        # puts "#{filename}, #{key}" if active
-        item_class = active ? 'nav-group active' : 'nav-group'
+    def write_sidebar_to_files(files)
```

Comment:
Codacy found an issue: [Assignment Branch Condition size for write_sidebar_to_files is too high. [<8, 22, 2> 23.49/15]](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-design-guide-jekyll-theme/pullRequest?prid=6198407)